namespaces/mson-namespace.md

```diff
@@ -0,0 +1,503 @@
+                }
+            ]
+        }
+    ]
+}
+```
+
+# MSON DOM Elements Definition
+
+## MSON Element (Distinct Element)
+
+Base element for all MSON elements.
+
+The MSON Element adds attributes representing MSON _Type Definition_ and _Type Sections_. Note in MSON DOM _Nested Member Types_ _Type Section_ is the `content` of the element.
+
+### Properties
+
+- `attributes`
+    - `typeAttributes` (array) - _Type Definition_ attributes list, see _Type Attribute_  
+        - (enum[string])
+            - required
+            - optional
+            - fixed
+            - sample
+            - default
+    - `variable` (boolean) - Element content is _Variable Value_
+    - `sample` (MSON Element) - Alternative sample _Member Types_ value
+    - `default` (MSON Element) - Default value for _Member Types_
+    - `validation` - Not used, reserved for a future use
+    - `description` - Combined description of an MSON Element
+
+## Boolean Type (JSON:Boolean Type)
+
+- Include MSON Element
+
+## String Type (JSON:String Type)
+
+- Include MSON Element
+
+## Number Type (JSON:Number Type)
+
+- Include MSON Element
+
+## Array Type (JSON:Array Type)
+
+- Include MSON Element
+
+## Object Type (JSON:Object Type)
+
+- Include MSON Element
+
+## Property Type (JSON:Property Type)
+
+- Include MSON Element
+
+## Enum Type (MSON Element)
+
+Enumeration type. Exclusive list of possible elements. The elements in content's array MUST be interpreted as mutually exclusive.
+
+### Properties
+
+- `element`: enum (string, required, fixed)
+- `content` (array[MSON Element])
+
+### Example
+
+#### MSON
+
+```
+- tag (enum[string])
+    - red
+    - green
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "property",
+            "attributes": {
+                "name": "tag"
+            },
+            "content": {
+                "element": "enum",
+                "content": [
+                    {
+                        "element": "string",
+                        "content": "red"
+                    },
+                    {
+                        "element": "string",
+                        "content": "green"
+                    }
+                ]
+            }
+        }
+    ]
+}
+```
+
+
+
+## Examples
+
+### Anonymous Object Type
+
+#### MSON
+
+```
+- id: 1
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "property",
+            "attributes": {
+                "name": "id"
+            },
+            "content": {
+                "element": "string",
+                "content": "1"
+            }
+        }
+    ]
+}
+```
+
+### Type Attributes
+
+#### MSON
+
+```
+- id: 42 (required, fixed)
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "property",
+            "attributes": {
+                "name": "id",
```

Comment:
Did you mean to replace `name` by this? https://github.com/refractproject/refract-spec/pull/4/files#diff-925248c117ee875155bdb1606d4fd897R90